resp.go

```diff
@@ -77,12 +77,16 @@ func init() {
 }
 
 func readSimpleString(i *bufio.Reader) (m RedisMessage, err error) {
-	m.string, err = readS(i)
+	var s string
+	s, err = readS(i)
+	m.setString(s)
```

Comment:
Using `readS` first and then `setString` seems a little redundant. Can we make `readS` return the length and the unsafe.SliceData directly? Same to the `readB`, `readA`, and `readE`.